src/lib/address/yat.ts

```diff
@@ -37,19 +39,43 @@ export const validateYat: ValidateYat = ({ value }) => {
 }
 
 export const resolveYat: ResolveYat = async args => {
+  const { value, assetId } = args
   try {
     const { data } = await axios.get<YatResponse>(
-      `${getConfig().REACT_APP_YAT_NODE_URL}/emoji_id/${args.value}`,
+      `${getConfig().REACT_APP_YAT_NODE_URL}/emoji_id/${value}`,
     )
     if (data.error) return ''
 
-    const found = data.result.find(
-      emoji => emoji.tag === '0x1004', // 0x1004 is eth address
-    )
+    // see https://a.y.at/emoji_id/%F0%9F%A6%8A%F0%9F%9A%80%F0%9F%8C%88 for example
+    // https://api-docs.y.at/docs/sdks/nodejs/docs/emojiidapi#lookupemojiidpayment
+    /**
+     * y.at allows different addresses to be mapped to different assets
+     * this hands the case of a USDC address and an ETH address
+     */
+    const USDC_TAG_ID = '0x6300'
+    const ETH_TAG_ID = '0x1004'
+    const maybeUSDCData = data.result.find(emoji => emoji.tag === USDC_TAG_ID)
+    const maybeETHData = data.result.find(emoji => emoji.tag === ETH_TAG_ID)
+
+    const USDC_ASSET_ID: AssetId = 'eip155:1/erc20:0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48'
+
+    const maybeAddress = (() => {
+      switch (assetId) {
+        case USDC_ASSET_ID: {
+          if (!maybeUSDCData) return ''
```

Comment:
We may want to fallback to the ETH address:
> If a yat has no USDC address mapped, attempts to send USDC resolve to the ETH address if one exists

Reply:
ahhh good point

Reply:
https://github.com/shapeshift/web/pull/4153/commits/4f825ed63d0eb90baf3b61f5034ef9a4069f1ea3